index.js

```diff
@@ -179,21 +179,42 @@ program
 
         if (onePagePath) {
           const lazyReloadMiddleware = function (req, res, next) {
-            const isHtmlFile = req.url.endsWith('.html');
+            const requestExtension = path.posix.extname(req.url);
 
-            if (isHtmlFile) {
-              const isDynamicIncludeHtmlFile = req.url.endsWith('._include_.html');
+            const hasEndingSlash = req.url.endsWith('/');
+            const hasNoExtension = requestExtension === '';
+            const isHtmlFileRequest = requestExtension === '.html' || hasEndingSlash || hasNoExtension;
 
-              if (!isDynamicIncludeHtmlFile) {
-                const urlWithoutBaseUrl = req.url.replace(config.baseUrl, '');
-                const urlWithoutExtension = fsUtil.removeExtension(urlWithoutBaseUrl);
+            if (!isHtmlFileRequest || req.url.endsWith('._include_.html')) {
+              next();
+              return;
+            }
 
-                const didInitiateRebuild = site.changeCurrentPage(urlWithoutExtension);
-                if (didInitiateRebuild) {
-                  req.url = ensurePosix(path.join(config.baseUrl || '/', LAZY_LOADING_SITE_FILE_NAME));
-                }
+            if (hasNoExtension && !hasEndingSlash) {
+              // Urls of type 'host/userGuide' - check if 'userGuide' is a raw file or does not exist
+              const diskFilePath = path.resolve(rootFolder, req.url);
+              if (!fs.existsSync(diskFilePath) || !fs.isDirectorySync(diskFilePath)) {
+                // Request for a raw file
+                next();
+                return;
               }
             }
+
+            // Remove baseUrl
+            let normalizedUrl = req.url.replace(config.baseUrl, '');
+
+            // Map 'hostname/userGuide/' and 'hostname/userGuide' to hostname/userGuide/index.
+            normalizedUrl = (hasNoExtension || hasEndingSlash)
```

Comment:
Should we do:
```
const urlWithoutBase = ...
const urlWithIndex = ...
const urlWithoutExtension = ...
```

I think we should avoid mutation where possible. In this case, we can break this transformation down into individual independent steps with variables names that make each step self-explanatory.

Reply:
The logical flow is fairly straightforward here, but in other more complex cases variable mutation can lead to some hard to trace errors. I have a strong preference for `const` over `let` whenever possible to avoid this, but I'm happy to discuss if you (or anyone else) has a different preference :)